Mathlib/Algebra/Central/TensorProduct.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2025 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Yunzhou Xie
+-/
+
+import Mathlib.Algebra.Central.Basic
+import Mathlib.RingTheory.Flat.FaithfullyFlat.Basic
+import Mathlib.LinearAlgebra.FiniteDimensional.Defs
+
+/-!
+
+## Main Results
+
+- `Algebra.IsCentral.left_of_tensor`: If `B` and `C` are nontrivial finite dimensional
+  `K`-algebras, and `B ⊗[K] C` is a central algebra over `K`, then `B` is a
+  central algebra over `K`.
+- `Algebra.IsCentral.right_of_tensor`: If `B` and `C` are nontrivial finite dimensional
+  `K`-algebras, and `B ⊗[K] C` is a central algebra over `K`, then `C` is a
+  central algebra over `K`.
+
+## Tags
+Central Algebras, Central Simple Algebras, Noncommutative Algebra
+-/
+
+universe u v
+
+open TensorProduct
+
+variable (K : Type u) [Field K]
+
+namespace Algebra.IsCentral
+
+lemma left_of_tensor (B C : Type v)
+    [Ring B] [Ring C] [Nontrivial B] [Nontrivial C] [Algebra K C] [Algebra K B]
+    [FiniteDimensional K B] [hbc : Algebra.IsCentral K (B ⊗[K] C)] :
+    IsCentral K B := by
+  letI : Nontrivial (B ⊗[K] C) := Module.FaithfullyFlat.lTensor_nontrivial _ _ _
+  have : (Algebra.TensorProduct.includeLeft.comp (Subalgebra.center K B).val).range ≤
+    Subalgebra.center K (B ⊗[K] C) := fun x hx ↦ by
+    simp only [AlgHom.mem_range, AlgHom.coe_comp, Subalgebra.coe_val, Function.comp_apply,
+      Algebra.TensorProduct.includeLeft_apply, Subtype.exists, exists_prop] at hx
+    obtain ⟨b, hb0, hb⟩ := hx
+    rw [Subalgebra.mem_center_iff] at hb0 ⊢
+    intro bc
+    induction bc using TensorProduct.induction_on with
+    | zero => simp
+    | tmul b' c =>
+      subst hb
+      simp only [Algebra.TensorProduct.tmul_mul_tmul, mul_one, one_mul]
+      congr 1
+      exact hb0 b'
+    | add _ _ _ _ => simp_all [add_mul, mul_add]
+  have eq: (Algebra.TensorProduct.includeLeft.comp (Subalgebra.center K B).val).range =
```

Comment:
`(Subalgebra.center K B).map (Algebra.TensorProduct.includeLeft)` might be a better spelling compared to `LinearMap.range` (@alreadydone told me this in a previous PR)